back-end/src/main/java/index/Indexer.java

```diff
@@ -52,12 +54,24 @@ else if (Config.indexingScheme == Config.indexingScheme.TILE_INDEX)
     // precompute
     public static void precompute() throws Exception {
 
+        // return; 
+
         System.out.println("Precomputing...");
 
         associateIndexer();
-        for (Canvas c : Main.getProject().getCanvases())
-            for (int layerId = 0; layerId < c.getLayers().size(); layerId ++)
+        for (Canvas c : Main.getProject().getCanvases()) {
+            System.out.println("indexing canvas with id: " + c.getId());
+            System.out.println("canvas has num id: " + c.getNumericId());
+            for (int layerId = 0; layerId < c.getLayers().size(); layerId ++) {
                 c.getLayers().get(layerId).getIndexer().createMV(c, layerId);
+            }
+        }
+
+        if (Config.indexingScheme == Config.IndexingScheme.CUBE_INDEX) {
```

Comment:
as we discussed, indexing should all be done in `createMV`, so you don't need an additional indexing function. `create index` can be called only in the bottom level once you get #73 fixed.

Reply:
yes, i fixed this in my most recent commit, will update the issue with the fix